src/Umbraco.Web.UI.Client/src/common/services/contenteditinghelper.service.js

```diff
@@ -506,6 +506,39 @@ function contentEditingHelper(fileManager, $q, $location, $routeParams, editorSt
 
         },
 
+        /**
+         * @ngdoc method
+         * @name umbraco.services.contentEditingHelper#getPermissionsForContent
+         * @methodOf umbraco.services.contentEditingHelper
+         * @function
+         *
+         * @description
+         * Returns a object with permissions for user.
+         */
+        getPermissionsForContent: function () {
+
+          let currentNodePermissions = null;
+
+          // Just ensure we do have an editorState
+          if (editorState.current) {
+            // Fetch current node allowed actions for the current user
+            // This is the current node & not each individual child node in the list
+            const currentUserPermissions = editorState.current.allowedActions || [];
+
+            // Create a nicer model rather than the funky & hard to remember permissions strings
+            currentNodePermissions = {
+              "canCopy": currentUserPermissions.includes('O'), //Magic Char = O
+              "canCreate": currentUserPermissions.includes('C'), //Magic Char = C
+              "canDelete": currentUserPermissions.includes('D'), //Magic Char = D
+              "canMove": currentUserPermissions.includes('M'), //Magic Char = M
+              "canPublish": currentUserPermissions.includes('U'), //Magic Char = U
+              "canUnpublish": currentUserPermissions.includes('Z') //Magic Char = Z
```

Comment:
The original function had this:

```
"canUnpublish": _.contains(currentUserPermissions, 'U') //Magic Char = Z (however UI says it can't be set, so if we can publish 'U' we can unpublish)
```

Alternatively we could have:

```
"canUnpublish": currentUserPermissions.includes('Z') || currentUserPermissions.includes('U')
```

Reply:
I tweaked that function a little to return early if no editorState, which then reduces indentation and helps readability (a tiny amount). 

I think the current approach is good - one MagicChar for each action, it's clear and easier to understand.